homeassistant/components/totalconnect/__init__.py

```diff
@@ -1 +1,61 @@
 """The totalconnect component."""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Total Connect'
+
+DOMAIN = 'totalconnect'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+TOTALCONNECT_PLATFORMS = ['alarm_control_panel']
+
+
+class TotalConnectSystem:
+    """TotalConnect System class."""
+
+    def __init__(self, username, password):
+        """Initialize the TotalConnect system."""
+        from total_connect_client import TotalConnectClient
+
+        _LOGGER.debug("Setting up TotalConnectSystem...")
+        self._username = username
+        self._password = password
+        self._client = TotalConnectClient.TotalConnectClient(username,
```

Comment:
Please create the client instance outside of init method and pass the client in as a parameter to this class. We try to avoid side effects in init method.

Reply:
The client should validate that the credentials are correct and if not, we should log error and `return False` from `setup`.